include/atheme/hook.h

```diff
@@ -238,6 +238,13 @@ struct hook_user_req
 	struct mynick *     mn;
 };
 
+struct hook_user_certfp
+{
+	struct sourceinfo * si;
+	struct myuser *     mu;
+	char                certfp[512];
```

Comment:
Instead of a fixed-sized allocation here, we could have this field be a `char *` to a `sstrdup`d copy of the fingerprint. If someone wanted to resize it up or down they'd be able to replace the pointer with a new one (making it bigger seems unlikely, but would work). We could free whatever happened to be in this field after calling the hook. (If a hook happened to replace it with a NULL and took ownership of what used to be there, no big deal)

Reply:
That'd also work, but involves an extra heap allocation. I'm not sure there's much benefit to it in this case; practically, there's a length limit for fingerprints anyway for them to be usable via the IRC protocol framing (and in `CERT LIST` output).

`512` does feel like a bit of a magic number here, though I'm not sure there's a good constant defined for it – we don't specify a size limit for certfps, and `BUFSIZE` seems slightly excessive as the value needs to fit within (less than) 512 bytes for it to be usable in any case

Reply:
I'm fine with `512` here, and any registered hooks that wanted to modify the fingerprint would usually have to copy it to a temporary buffer and then write it back anyway, at which point they run into the same considerations.